### PR TITLE
Update package structure

### PR DESCRIPTION
**Pull request by thisac**
An effort to make minorminer more Ocean-like.
* Removes Matlab support (deprecated since 0.2.8)
* Removes unnecessary function imports
* Remove git hook scripts
* Update package structure to be more like the rest of Ocean
* Move `test_requirements.txt` to `tests/requirements.txt`

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -7,9 +7,12 @@ requires = [
 build-backend = "setuptools.build_meta"
 
 [project]
+name = "minorminer"
+dynamic = ["version"]
 authors = [
-    {name = "Kelly Boothby", email = "boothby@dwavesys.com"},
```
Comment (arcondello) [inline pyproject.toml:11]: IMO we should do
```.toml
authors = [
    {name = "Kelly Boothby"},
]
maintainers = [
    {name = "D-Wave Inc.", email = "tools@dwavesys.com"},
]
```

**Review by arcondello — COMMENTED** [review]
I missed that we started using `reno` in this package. We should add it to the `requirements.txt` and to the README under contributing.

File: releasenotes/notes/remove-matlab-support-61797b65ff5bfa5a.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+---
+upgrade:
+  - Matlab support was deprecated as of vesion 0.2.8 and removed in the current version.
```
Comment (randomir) [inline releasenotes/notes/remove-matlab-support-61797b65ff5bfa5a.yaml]:
```suggestion
  - Matlab support was deprecated as of version 0.2.8 and removed in the current version.
```

**Review by randomir — APPROVED** [review]
LGTM

File: pyproject.toml (hunk 1 above)
Comment (thisac) [reply pyproject.toml:11]: Happy to go with that; makes sense to me.

**Review by arcondello — APPROVED** [review]

**Pull request merged**